packages/build/src/utils/semver.js

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+const { major, minor, patch, clean: cleanVersion } = require('semver')
+
+// According to semver, the second number is the major release number for
+// `0.*.*` versions and the third for `0.0.*`. This is how `^` behaves with the
+// `semver` module which is used by `npm`.
+const getMajorVersion = function (version) {
+  if (!version || cleanVersion(version) === null) {
+    return
+  }
+
+  const majorVersion = major(version)
+  if (majorVersion !== 0) {
+    return `${majorVersion}`
+  }
+
+  const minorVersion = minor(version)
+  if (minorVersion !== 0) {
+    return `${majorVersion}.${minorVersion}`
+  }
+
+  const patchVersion = patch(version)
+  return `${majorVersion}.${minorVersion}.${patchVersion}`
+}
+
+module.exports = { getMajorVersion }
```

Comment:
I guess this answers https://github.com/netlify/build/pull/2756#discussion_r628105036 😄